src/components/InterestForm/index.tsx

```diff
@@ -2,72 +2,94 @@
 
 import { useState, useEffect } from 'react';
 import { upsertInterest } from '@/api/supabase/queries/interest';
-import { Interest, CaseListing, RoleEnum } from '@/types/schema';
+import { Interest, Listing } from '@/types/schema';
 import { useAuth } from '@/utils/AuthProvider';
 import { isValidDate } from '@/utils/helpers';
 import { P, H3 } from '@/styles/text';
 import COLORS from '@/styles/colors';
-import DateInput from '../DateInput';
-import Button from '../Button';
-import {
-  FormContainer,
-  FormTextArea,
-  FormQuestion,
-  RadioGroup,
-  Radio,
-  RadioInput,
-  FormFooter,
-  FormWarning,
-  ErrorText,
-  RadioLabel,
-} from './styles';
+import { Flex } from '@/styles/containers';
+import RadioGroup from '@/components/RadioGroup';
+import TextAreaInput from '@/components/TextAreaInput';
+import DateInput from '@/components/DateInput';
+import Button from '@/components/Button';
+import { FormContainer, FormFooter, FormWarning } from './styles';
 
-const radioOptions = [
-  'Attorney',
-  'Interpreter',
-  'Either Attorney or Interpreter',
-];
-
-export default function InterestForm({ caseData }: { caseData: CaseListing }) {
+export default function InterestForm({
+  listingData,
+  interpretation = false,
+}: {
+  listingData: Listing;
+  interpretation?: boolean;
+}) {
   const auth = useAuth();
   const [reason, setReason] = useState<string>('');
-  const [rolesInterested, setRolesInterested] = useState<string>('');
+  const [needsInterpreter, setNeedsInterpreter] = useState<string>('');
   const [startDate, setStartDate] = useState<string>('');
   const [submitted, setSubmitted] = useState(false);
   const [missingInfo, setMissingInfo] = useState(false);
 
   useEffect(() => {
     // Reset form fields when caseData changes
     setReason('');
+    setNeedsInterpreter('');
     setStartDate('');
-    setRolesInterested('');
     setSubmitted(false);
     setMissingInfo(false);
-  }, [caseData]);
+  }, [listingData]);
 
   const handleInsert = async () => {
-    if (startDate === '' || rolesInterested === '') {
+    // Error handling
+    if (
+      (listingData.listing_type === 'CASE' &&
+        !interpretation &&
+        (startDate === '' || needsInterpreter === '')) ||
+      (((listingData.listing_type === 'CASE' && interpretation) ||
+        listingData.listing_type === 'INT') &&
+        startDate === '') ||
+      ((listingData.listing_type === 'LCA' ||
+        listingData.listing_type === 'DOC') &&
+        reason === '')
+    ) {
       setMissingInfo(true);
       return;
     }
-    if (isValidDate(startDate)) {
+    let responses = {};
```

Comment:
let's make this `const` and modify just the specific fields later
i.e. `responses.needs_interpreter: needsInterpreter === 'Yes'`

Reply:
done!